src/libutil/util.cc

```diff
@@ -54,6 +54,18 @@ std::optional<std::string> getEnv(const std::string & key)
     return std::string(value);
 }
 
+std::optional<std::string> getEnvNonEmpty(const std::string & key) {
+    auto value = getEnv(key);
+    if (value == "") {
+        // TODO: determine whether this should be a warning or an error.
+        logWarning({
+                .msg = hintfmt("ignoring the '%1%' env variable, its value has been set to \"\"", key)
+         });
```

Comment:
```suggestion
        warn("ignoring the '%1%' env variable, its value has been set to \"\"", key);
```

(unless you have a good reason for using `logWarning`, but I don't see it)